src/frontend/src/icp/services/worker.icrc-wallet.services.ts

```diff
@@ -42,8 +51,14 @@ export const initIcrcWalletWorker = async ({
 			case 'syncIcrcWalletError':
 				onLoadTransactionsError({
 					tokenId,
-					error: (data.data as PostMessageDataResponseError).error
+					error: (data.data as PostMessageDataResponseError).error,
+					silent: true
 				});
+
+				// In case of error, we start the listener again, but only with the ledgerCanisterId,
+				// to make it request only the balance and not the transactions
+				restartWorkerWithLedgerOnly();
```

Comment:
@peterpeterparker as discussed; do you think it is too much to have an internal function? I would not externalize it, just because I need to pass all props (worker, ledger, env)